com.ibm.streamsx.hdfs/impl/java/src/com/ibm/streamsx/hdfs/HDFS2FileSink.java

```diff
@@ -755,7 +789,7 @@ public void processPunctuation(StreamingInput<Tuple> arg0, Punctuation punct)
 		}
 		// set the file to expire after punctuation
 		// on the next write, the file will be recreated
-		else if (isCloseOnPunct() && fFileToWrite != null) {
```

Comment:
Why did you get rid of the `fFileToWrite != null`?  I believe that was there to prevent empty files being written when there are two puncutations with no tuples in between.